lib/new_relic/agent/instrumentation/grpc/client/instrumentation.rb

```diff
@@ -13,26 +13,48 @@ module GRPC
         module Client
           include NewRelic::Agent::Instrumentation::GRPC::Helper
 
-          def issue_request_with_tracing(method, requests, marshal, unmarshal,
+          def issue_request_with_tracing(grpc_type, method, requests, marshal, unmarshal,
             deadline:, return_op:, parent:, credentials:, metadata:)
             return yield unless trace_with_newrelic?
 
             segment = request_segment(method)
             request_wrapper = NewRelic::Agent::Instrumentation::GRPC::Client::RequestWrapper.new(@host)
             segment.add_request_headers(request_wrapper)
             metadata.merge!(metadata, request_wrapper.instance_variable_get(:@newrelic_metadata))
+            grpc_message = nil
+            grpc_status = 0
 
             NewRelic::Agent.disable_all_tracing do
-              NewRelic::Agent::Tracer.capture_segment_error(segment) do
```

Comment:
- For gRPC clients, go lower level when rescuing exceptions and use
  `notice_error` directly instead of `create_segment_error`.